### PR TITLE
feat(emqx_metrics): Sliding window samples

### PR DESCRIPTION
**Pull request by ieQu1**
Fixes [EMQX-6741](https://emqx.atlassian.net/browse/EMQX-6741).

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] For internal contributor: there is a jira ticket to track this change
- [x] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

### CODE REVIEW

**Comment by ieQu1** [comment]
The failed test is known to be flaky

File: apps/emqx/src/emqx_metrics_worker.erl [hunk 1]
```diff
@@ -126,14 +152,17 @@ child_spec(ChldName, Name) ->
         modules => [emqx_metrics_worker]
     }.
 
--spec create_metrics(handler_name(), metric_id(), [metric_name()]) -> ok | {error, term()}.
+-spec create_metrics(handler_name(), metric_id(), [metric_spec() | atom()]) -> ok | {error, term()}.
```
Comment (thalesmg) [inline apps/emqx/src/emqx_metrics_worker.erl:155]:
```suggestion
-spec create_metrics(handler_name(), metric_id(), [metric_spec() | metric_name()]) -> ok | {error, term()}.
```

File: apps/emqx/src/emqx_metrics_worker.erl [hunk 2]
```diff
@@ -126,14 +152,17 @@ child_spec(ChldName, Name) ->
...
 create_metrics(Name, Id, Metrics) ->
-    create_metrics(Name, Id, Metrics, Metrics).
+    Metrics1 = desugar(Metrics),
+    Counters = filter_counters(Metrics1),
+    create_metrics(Name, Id, Metrics1, Counters).
 
--spec create_metrics(handler_name(), metric_id(), [metric_name()], [metric_name()]) ->
+-spec create_metrics(handler_name(), metric_id(), [metric_spec() | atom()], [atom()]) ->
```
Comment (thalesmg) [inline apps/emqx/src/emqx_metrics_worker.erl:161]:
```suggestion
-spec create_metrics(handler_name(), metric_id(), [metric_spec() | metric_name()], [metric_name()]) ->
```

File: apps/emqx/src/emqx_metrics_worker.erl [hunk 3]
```diff
@@ -103,9 +116,22 @@
...
+    %% Total number of samples through the history
+    n_samples = 0 :: non_neg_integer(),
+    datapoints = [] :: [#slide_datapoint{}]
+}).
+
 -record(state, {
     metric_ids = sets:new(),
-    rates :: undefined | #{metric_id() => #rate{}}
+    rates :: #{metric_id() => #{atom() => #rate{}}} | undefined,
+    slides = #{} :: #{metric_id() => #{atom() => #slide{}}}
```
Comment (thalesmg) [inline apps/emqx/src/emqx_metrics_worker.erl:134]:
```suggestion
    rates :: #{metric_id() => #{metric_name() => #rate{}}} | undefined,
    slides = #{} :: #{metric_id() => #{metric_name() => #slide{}}}
```

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — DISMISSED** [review]

**Pull request merged**